### PR TITLE
Update VMs to the latest versions

### PR DESCRIPTION
**Pull request by ltratt**
This PR updates various VMs to the latest versions. It seems to work on OpenBSD and Linux _except_ that on both platforms, at the end, it does:

```
cd startup_runners && \
        env LD_LIBRARY_PATH=/home/ltratt/warmup_experiment/work/gcc-inst/lib CC=/home/ltratt/warmup_experiment/work/gcc-inst/bin/zgcc \
        JAVA_CPPFLAGS='"-I/home/ltratt/warmup_experiment/work/openjdk/build/bsd-x86_64-normal-server-release/images/j2sdk-image/include \
        -I/home/ltratt/warmup_experiment/work/openjdk/build/bsd-x86_64-normal-server-release/images/j2sdk-image/include/linux"' \
        JAVA_LDFLAGS=-L/home/ltratt/warmup_experiment/work/openjdk/build/bsd-x86_64-normal-server-release/images/j2sdk-image/lib \
        JAVAC=/home/ltratt/warmup_experiment/work/openjdk/build/bsd-x86_64-normal-server-release/jdk/bin/javac ENABLE_JAVA=1 gmake
gmake[1]: Entering directory '/home/ltratt/warmup_experiment/startup_runners'
/home/ltratt/warmup_experiment/work/gcc-inst/bin/zgcc -fPIC -Wall -Wextra -pedantic  -L/home/ltratt/warmup_experiment/startup_runners/../krun/libkruntime \
         outer_startup_runner.c \
        -o outer_startup_runner_c -lkruntime  \

/usr/bin/ld: cannot find -lkruntime
```

I haven't investigated this yet, but it's not obvious to me that it's a result of this PR, so maybe this is a longer-standing issue? Or ...?

### CODE REVIEW

**Comment by vext01** [comment]
Ah yes, I do know about this, but I had forgotted to raise a bug of fix it.

The startup experiment (that Carl made), has bit-rotted somehow. I first noticed this last week, and disabled this target on the last real run (since we don't yet use it).

bug raised, #125

**Comment by ltratt** [comment]
OK then, if you're OK with the rest of the PR, let's merge this and fix that bug in #125.

File: build.sh [hunk 1]
```diff
@@ -165,22 +165,30 @@ build_luajit() {
...
+        case `uname` in
+            OpenBSD) ftp $url || exit $?;;
+            *) wget $url || exit $?;;
+        esac
     fi
 
     if ! [ -d "${wrkdir}/pypy" ]; then
-        bunzip2 -c - pypy-${PYPYV}-src.tar.bz2 | tar xf -
-        mv pypy-${PYPYV}-src pypy
+        bunzip2 -c - pypy2-v${PYPYV}-src.tar.bz2 | tar xf -
```
Comment (vext01) [inline build.sh]: Might aswell add `|| exit $?` while we are here.

**Comment by vext01** [comment]
Tiny comment, but otherwise I don't see any clangers.

**Comment by ltratt** [comment]
Squash or merge as-is? [I'm fine either way.]

**Comment by vext01** [comment]
Probably OK as is.

**Comment by vext01** [comment]
Thankyou muchly

**Pull request merged**